src/Bicep.Core/TypeSystem/DeployTimeConstantViolationVisitor.cs

```diff
@@ -95,5 +101,12 @@ protected void FlagDeployTimeConstantViolation(SyntaxBase errorSyntax, DeclaredS
 
             return accessiblePropertyNames;
         }
+
+        public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
+        {
+            insideNameofFunction = syntax.Name.IdentifierName == LanguageConstants.NameofFunctionName;
+            base.VisitFunctionCallSyntax(syntax);
+            insideNameofFunction = false;
```

Comment:
Would it have the same effect if you short-circuited instead, i.e.:

```suggestion
            if (syntax.Name.IdentifierName == LanguageConstants.NameofFunctionName)
            {
                return;
            }

            base.VisitFunctionCallSyntax(syntax);
```

Reply:
not sure. This part I wrote in February and I don't remember now why ;) Will check that

Reply:
Yep, short-circuit would have same effect, fixed.